solutions/usaco-736.mdx

```diff
@@ -188,3 +188,75 @@ public class BovineGenomics {
 </LanguageSection>
 
 
+## Solution 3 - Unordered Map
+
+<LanguageSection>
+
+<CPPSection>
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(M * N)$
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int n, m, ans = 0;
+vector<string> gens;
+
+// Use `unordered_map` to achieve O(1) time complexity in queries
+unordered_map<char, int> spotties;
```

Comment:
use array, since you're storing chars. unordered map is sometimes slow (in this case, it won't matter, but in general it's a good idea to use array whenever possible)

Reply:
How is it possible to be slow?
https://github.com/gibsjose/cpp-cheat-sheet/blob/master/Data%20Structures%20and%20Algorithms.md#15-map-stdmap-and-stdunordered_map

Reply:
constant factor

Reply:
i honestly disagree with michael here- i'd recommend using a map instead of an array

Reply:
here its fine. but there's times where you TLE with map and passes with array

Reply:
those only happen at and above gold tbh

Reply:
in general its good to teach that you shouldnt use maps if not needed so you dont die to it in the future

i've died enough times to say this :P

Reply:
ok yeah true
but why not vectors then?

Reply:
thats ok too, idrc